logstash-core/src/main/java/org/logstash/ackedqueue/TailPage.java

```diff
@@ -14,19 +14,14 @@ public TailPage(HeadPage page) {
     }
 
     // create a new TailPage object for an exiting Checkpoint and data file
+    // @param @pageIO is expected to be open/recover/create
     public TailPage(Checkpoint checkpoint, Queue queue, PageIO pageIO) throws IOException {
         super(checkpoint.getPageNum(), queue, checkpoint.getMinSeqNum(), checkpoint.getElementCount(), checkpoint.getFirstUnackedSeqNum(), new BitSet(), pageIO);
 
         // this page ackedSeqNums bitset is a new empty bitset, if we have some acked elements, set them in the bitset
         if (checkpoint.getFirstUnackedSeqNum() > checkpoint.getMinSeqNum()) {
             this.ackedSeqNums.flip(0, (int) (checkpoint.getFirstUnackedSeqNum() - checkpoint.getMinSeqNum()));
         }
-
-        if (pageIO != null) {
-            // open the data file and reconstruct the IO object internal state
-            pageIO.open(checkpoint.getMinSeqNum(), checkpoint.getElementCount());
```

Comment:
the only reason we are safe in removing this is because this is the only place we construct?

https://github.com/elastic/logstash/pull/6519/files#diff-69b2c41a7ba6596afa233505525212bbR169